### ISSUE TITLE
PyBake Commands Should Be More Modular

### ISSUE THREAD

**Issue by Manuzor** [lab132/PyBake#20]
Right now, everything is in `__main__`. This should be split out to something more modular, so that adding new parsers later on is easier.

I suggest using a decorator (e.g. `command`) which registers some Command class somewhere. Example:

~~~{.py}
# Decorate a class
@command("oven")
class OvenCommand:
  def __init__(self):
    self.help = "bla bla."
    ...

# Or decorate a function (same semantics as above, for the caller)
@command("oven")
def ovenCommand():
  return OvenCommand()
~~~

I also suggest putting all commands to a sub directory commands, i.e. `PyBake.commands` and e.g. `PyBake.commands.oven`, potentially splitting the _command_ from the actual logic of the command, which would still reside in `PyBake.<command>`. If we do this dynamically enough, the user might be able to supply some custom commands by specifying one or more directories in which to look for command modules, similar to what `git` does with everything in the `PATH` that starts with `git-`.

@pampersrocker Since you're already working on it on the `split-main` branch, I'm assigning this to you. :smile:

**Comment by pampersrocker** [comment]
I've started moving the parser generator in split-main as you already noticed by creating a "ModuleManager" in each submodule see https://github.com/lab132/PyBake/blob/split-main/PyBake/depot.py#L10 for example

**Comment by pampersrocker** [comment]
Submodule `commands` has been created in 4b6efd1

**Comment by pampersrocker** [comment]
Merged in 5431536
